### PR TITLE
Test on Ubuntu 24.04 GitHub Images

### PR DESCRIPTION
**Pull request by nibanks**
## Description

Just noticed that GitHub has beta support for Ubuntu 24.04. Let's test it out and see how it goes.

## Testing

CI/CD

## Documentation

N/A

### CODE REVIEW

**Comment by ami-GS** [comment]
https://github.com/PowerShell/PowerShell/issues/21385

File: .docker/ubuntu-24.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Stage 1: Base image
```
Comment (nibanks) [inline .docker/ubuntu-24.04/Dockerfile:1]: should we push this to a separate PR first, so we can then use this PR to validate it?

**Comment by ami-GS** [comment]
There are build errors in several flavors.

File: src/generated/stdout/quic_trace.c [hunk 2]
```diff
@@ -19,9 +19,9 @@
 #include <quic_trace.h>
 
 
-void EncodeHexBuffer(_In_reads_(BufferLen) uint8_t * Buffer,
```
Comment (nibanks) [inline src/generated/stdout/quic_trace.c:22]: I rather ignore this error than make this change. This code was specifically designed for small buffers where size fit in `uint8_t`.

**Comment by ami-GS** [comment]
`-CodeCheck` on Ubuntu 24.04 (GCC 13.2.0) shows error below. This cannot be suppressed by `add_compile_options(-Wno-misc-header-include-cycle)` in `src/submodules/CMakeLists.txt` or `src/generated/CMakeLists.txt`


```
[ 94%] Built target msquic.lttng
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/macros.h:15:10: error: circular header file dependency detected while including 'opensslv.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   15 | #include <openssl/opensslv.h>
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/opensslv.h:109:11: note: 'opensslv' included from here
  109 | # include <openssl/macros.h>
      |           ^
/root/msquic/src/platform/crypt_openssl.c:14:10: note: 'opensslv.h' included from here
   14 | #include "openssl/opensslv.h"
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/srtp.h:25:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   25 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:970:11: note: 'ssl.hs' included from here
  970 | # include <openssl/srtp.h>      /* Support for the use_srtp extension */
      |           ^
/root/msquic/src/platform/crypt_openssl.c:38:10: note: '<U+0006><U+0006><U+0006><U+0006><U+0006>' included from here
   38 | #include "openssl/ssl.h"
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl3.h:23:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   23 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:967:11: note: 'ssl.hr' included from here
  967 | # include <openssl/ssl3.h>
      |           ^
/root/msquic/src/platform/crypt_openssl.c:38:10: note: 'ssl.h' included from here
   38 | #include "openssl/ssl.h"
      |          ^
354 warnings generated.
Suppressed 351 warnings (349 in non-user code, 2 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
3 warnings treated as errors
gmake[2]: *** [src/platform/CMakeFiles/platform.dir/build.make:286: src/platform/CMakeFiles/platform.dir/crypt_openssl.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/srtp.h:25:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   25 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:970:11: note: 'ssl.hs' included from here
  970 | # include <openssl/srtp.h>      /* Support for the use_srtp extension */
      |           ^
/root/msquic/src/platform/selfsign_openssl.c:29:10: note: '<U+0006><U+0006><U+0006><U+0006><U+0006>' included from here
   29 | #include "openssl/ssl.h"
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl3.h:23:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   23 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:967:11: note: 'ssl.hr' included from here
  967 | # include <openssl/ssl3.h>
      |           ^
/root/msquic/src/platform/selfsign_openssl.c:29:10: note: 'ssl.h' included from here
   29 | #include "openssl/ssl.h"
      |          ^
368 warnings generated.
Suppressed 366 warnings (363 in non-user code, 3 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
2 warnings treated as errors
gmake[2]: *** [src/platform/CMakeFiles/platform.dir/build.make:314: src/platform/CMakeFiles/platform.dir/selfsign_openssl.c.o] Error 1
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/macros.h:15:10: error: circular header file dependency detected while including 'opensslv.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   15 | #include <openssl/opensslv.h>
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/opensslv.h:109:11: note: 'opensslv' included from here
  109 | # include <openssl/macros.h>
      |           ^
/root/msquic/src/platform/tls_openssl.c:14:10: note: 'opensslv.h' included from here
   14 | #include "openssl/opensslv.h"
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/srtp.h:25:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   25 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:970:11: note: 'ssl.hs' included from here
  970 | # include <openssl/srtp.h>      /* Support for the use_srtp extension */
      |           ^
/root/msquic/src/platform/tls_openssl.c:35:10: note: '<U+0006><U+0006><U+0006><U+0006><U+0006>' included from here
   35 | #include "openssl/ssl.h"
      |          ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl3.h:23:11: error: circular header file dependency detected while including 'ssl.h', please check the include path [misc-header-include-cycle,-warnings-as-errors]
   23 | # include <openssl/ssl.h>
      |           ^
/root/msquic/build/linux/x64_openssl3/_deps/opensslquic-build/openssl3/include/openssl/ssl.h:967:11: note: 'ssl.hr' included from here
  967 | # include <openssl/ssl3.h>
      |           ^
/root/msquic/src/platform/tls_openssl.c:35:10: note: 'ssl.h' included from here
   35 | #include "openssl/ssl.h"
      |          ^
486 warnings generated.
Suppressed 484 warnings (481 in non-user code, 1 NOLINT, 2 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
3 warnings treated as errors
gmake[2]: *** [src/platform/CMakeFiles/platform.dir/build.make:272: src/platform/CMakeFiles/platform.dir/tls_openssl.c.o] Error 1
gmake[2]: Leaving directory '/root/msquic/build/linux/x64_openssl3'
gmake[1]: *** [CMakeFiles/Makefile2:294: src/platform/CMakeFiles/platform.dir/all] Error 2
gmake[1]: Leaving directory '/root/msquic/build/linux/x64_openssl3'
gmake: *** [Makefile:146: all] Error 2
```

**Comment by ami-GS** [comment]
Remaining is android issue. I have no idea the cause
image

**Comment by nibanks** [comment]
Can we just disable Android on 24.04 for now then?

**Review by ProjectsByJackHe — APPROVED** [review]
LGTM, as long as tests are passing, and this looks to be mostly pipeline scripting.

**Pull request merged**